src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -122,6 +126,17 @@ protected override ILspServices ConstructLspServices()
 
         services.AddSingleton<IFilePathService, LSPFilePathService>();
 
+        if (_projectInfoDriver is { } projectInfoDriver)
+        {
+            services.AddSingleton<IRazorProjectInfoDriver>(_projectInfoDriver);
+        }
+        else
+        {
+            // If the language server was not created with an IRazorProjectInfoDriver,
+            // fall back to a FileWatcher-base driver.
+            services.AddSingleton<IRazorProjectInfoDriver, FileWatcherBasedRazorProjectInfoDriver>();
```

Comment:
I wonder if the `FileWatcherBasedRazorProjectInfoDriver` should be constructed by rzls, and passed in to here?

Reply:
I had wanted to do that, but it needs the IWorkspaceRootPathProvider, which is tied to the CapabilitiesManager. I'll take another look.

Reply:
I took another look. Unfortunately, we just don't know what the root path is when `rzls` is constructed. Since `FileWatcherBasedRazorProjectInfoDriver` is dependent on `IWorkspaceRootPathProvider`, and that's dependent on the `CapabilitiesManager`, it really needs to be constructed by DI. Maybe we can improve this later, but I think it's probably OK for now.